sentry-core/src/main/java/io/sentry/core/EventCachedEventFireAndForgetIntegration.java

```diff
@@ -0,0 +1,125 @@
+package io.sentry.core;
+
+import static io.sentry.core.ILogger.logIfNotNull;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.nio.charset.Charset;
+import java.util.concurrent.Executors;
+
+public final class EventCachedEventFireAndForgetIntegration implements Integration {
+  private static final Charset UTF_8 = Charset.forName("UTF-8");
+
+  @Override
+  public void register(IHub hub, SentryOptions options) {
+    String cachedDir = options.getCacheDirPath();
+    if (cachedDir == null) {
+      logIfNotNull(
+          options.getLogger(), SentryLevel.WARNING, "No cache dir path is defined in options.");
+      return;
+    }
+
+    File outbox = new File(options.getCacheDirPath());
+    try {
+      Executors.callable(
+              () -> {
+                SendCachedFiles(hub, options.getLogger(), options.getSerializer(), outbox);
```

Comment:
why don't you have a global variable caching what we need to use in this class in different methods, like `ILogger, ISerializer` and so on, just easier than passing them around, or just the `SentryOptions`, we are doing this way in the other integrations

Reply:
The problem with `SentryOptions` for me is that it is mutable - this enables logging to be changed at runtime, which is good, but I'm not sure swapping out the serializer (or cache if it were returned from the options) is something we want to support/enable. So while you can set the serializer on the options it doesn't affect the client that already exists - that client will always use the serializer that it was instantiated with. This eliminates problems like inability to deserialize an event after it was successfully serialized using a different serializer.

Reply:
What @metlos said and also, I want to make it easier to pull the whole logic out of this class so it can be reused not only from this "FireAndForget" approach.